### PR TITLE
Changes for payment account creation flow

### PR DESCRIPTION
**Pull request by saravanpa-aot**
*Issue #:*
https://github.com/bcgov/entity/issues/5098

*Description of changes:*

- Major changes to remove payment_type from auth. Corresponding endpoints are removed

- Create payment type in Pay database on creation of account

- Frontend changes to read bcol details from org instead of payment endpoint

- In the rest service added a method to fetch service account token

- the change in product.py is because , it was breaking the transaction.

*Checklist:*
I confirm that below checklist items are addressed in this pull request; (check the boxes applicable)
- [ ] No lint errors
- [ ] No test case failures and proper coverage for all modules/classes
- [ ] Updated the deployment configs for new environment variable(s)
- [ ] Updtaed the postman collection in entity repository (https://github.com/bcgov/entity/tree/master/api-e2e/postman) for e2e tests

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the sbc-auth license (Apache 2.0).

### CODE REVIEW

File: auth-api/src/auth_api/models/__init__.py [hunk 1]
```diff
@@ -21,7 +21,7 @@
 from sbc_common_components.tracing.db_tracing import DBTracing  # noqa: I001, I004
 
 from .account_login_options import AccountLoginOptions
-from .account_payment_settings import AccountPaymentSettings
+from .account_payment_settings import AccountPaymentSettings as AccountPaymentSettingsDeprecated
```
Comment (pwei1018) [inline auth-api/src/auth_api/models/__init__.py:24]: Why call 'deprecated'?

Comment (saravanpa-aot) [reply auth-api/src/auth_api/models/__init__.py:24]: hm..the model is no longer used... we removed the AccountPaymentSettings from auth side .. I need the model for migration ... 
just to make sure nobody else is planning to use that model...not really necessary or there could be better way to do this...i am open to ideas

**Review by pwei1018 — APPROVED** [review]

**Review by sumesh-aot — APPROVED** [review]

File: auth-api/src/auth_api/models/__init__.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (saravanpa-aot) [reply auth-api/src/auth_api/models/__init__.py:24]: yeah...this was the easiest way ;though ugly

File: auth-api/config.py [hunk 2]
```diff
@@ -165,6 +165,13 @@ class _Config(object):  # pylint: disable=too-few-public-methods
     # Config value to disable activity logs
     DISABLE_ACTIVITY_LOGS = os.getenv('DISABLE_ACTIVITY_LOGS', 'False').lower() == 'true'
 
+    # Service account details
+    KEYCLOAK_SERVICE_ACCOUNT_ID = os.getenv(('KEYCLOAK_SERVICE_ACCOUNT_ID'))
```
[the remark this answers is not in the document]
Comment (saravanpa-aot) [reply auth-api/config.py:169]: missed that.. it still somehow worked though..

[the remark this answers is not in the document]
Comment (saravanpa-aot) [reply auth-api/config.py:169]: hm.. no no..its just redundant parantheis..to make a tuple , need an extra comma..i just got lucky

**Review by nitheesh-aot — APPROVED** [review]

**Pull request merged**